src/main/java/org/jabref/gui/exporter/SaveDatabaseAction.java

```diff
@@ -161,6 +162,19 @@ boolean saveAs(Path file, SaveDatabaseMode mode) {
                 .build();
         Optional<Path> selectedPath = dialogService.showFileSaveDialog(fileDialogConfiguration);
         selectedPath.ifPresent(path -> preferences.setWorkingDirectory(path.getParent()));
+        if(selectedPath.isPresent()){
+            Path savePath = selectedPath.get();
+            // Workaround for linux systems not adding file extension
+            if(!(savePath.getFileName().toString().toLowerCase().endsWith(".bib"))){
+                savePath = Path.of(savePath.toString()+".bib");
+                if(Files.notExists(savePath)){
+                  selectedPath = Optional.of(savePath);
+                } else {
+                  selectedPath = Optional.empty();
+                  dialogService.notify(Localization.lang("Could not save file."));
```

Comment:
As a User I should have the option to confirm that I want to overwrite the file if it exists

Reply:
That seems like a pretty reasonable request. Do you have any recommendations on where to look for UI popups keeping in theme with JabRef? I haven't used much of Java's UI features previously. Any documentation or advice is much appreciated.

Reply:
Just look in Jabref's dialogService methods: dialogService.showConfirmationDialogAndWait(...)
You will find an example somewhere in the code